disk-buffering/README.md

```diff
@@ -0,0 +1,114 @@
+* For a MetricExporter, it must be wrapped within
+  a [MetricDiskExporter](src/main/java/io/opentelemetry/contrib/disk/buffering/exporters/MetricDiskExporter.java).
+* For a SpanExporter, it must be wrapped within
+  a [SpanDiskExporter](src/main/java/io/opentelemetry/contrib/disk/buffering/exporters/SpanDiskExporter.java).
+
+Each wrapper will need the following when instantiating them:
+
+* The exporter to be wrapped.
+* A File instance of the root directory where all the data is going to be written. The same root dir
+  can be used for all the wrappers, since each will create their own folder inside it.
+* An instance
+  of [StorageConfiguration](src/main/java/io/opentelemetry/contrib/disk/buffering/storage/StorageConfiguration.java)
+  with the desired parameters. You can create one with default values by
+  calling `StorageConfiguration.getDefault()`.
+
+After wrapping your exporters, you must register the wrapper as the exporter you'll use. It will
+take care of always storing the data it receives.
+
+#### Set up example for spans
+
+```java
+// Creating the SpanExporter of our choice.
+SpanExporter mySpanExporter = OtlpGrpcSpanExporter.getDefault();
+
+// Wrapping our exporter with its disk exporter.
+SpanDiskExporter diskExporter = SpanDiskExporter.create(mySpanExporter, new File("/my/signals/cache/dir"), StorageConfiguration.getDefault());
+
+ // Registering the disk exporter within our OpenTelemetry instance.
+SdkTracerProvider myTraceProvider = SdkTracerProvider.builder()
+        .addSpanProcessor(SimpleSpanProcessor.create(diskExporter))
+        .build();
+OpenTelemetrySdk.builder()
+        .setTracerProvider(myTraceProvider)
+        .buildAndRegisterGlobal();
+
+```
+
+### Reading data
+
+Each of the exporter wrappers can read from the disk and send the retrieved data over to their
+wrapped exporter by calling this method from them:
+
+```java
+try {
+    if(diskExporter.exportStoredBatch(1, TimeUnit.SECONDS)) {
+        // A batch was successfully exported and removed from disk. You can call this method for as long as it keeps returning true.
+    } else {
+        // Either there was no data in the disk or the wrapped exporter returned CompletableResultCode.ofFailure().
+    }
+} catch (IOException e) {
+    // Something unexpected happened.
+}
+```
+
+Both, the writing and reading processes can run in parallel and the way they don't overlap is
+because each is supposed to happen in different files, and the way it is ensured that they don't
+accidentally meet in the same file is by using the configurable parameters, which ensure to set time
+frames for each action to be done on a single file at a time. On top of that, there's a mechanism in
+place to avoid overlapping on edge cases where the time frames ended but the resources haven't been
+released, for that mechanism to work properly, this tool assumes that both, the reading and the
+writing actions, are executed within the same application process.
+
```

Comment:
```suggestion
Both the writing and reading processes can run in parallel and they don't overlap
because each is supposed to happen in different files. We ensure that reader and writer don't
accidentally meet in the same file by using the configurable parameters. These parameters set non-overlapping time frames for each action to be done on a single file at a time. On top of that, there's a mechanism in
place to avoid overlapping on edge cases where the time frames ended but the resources haven't been
released. For that mechanism to work properly, this tool assumes that both the reading and the
writing actions are executed within the same application process.

```

Reply:
Oh and this obviously isn't a show-stopper.